glusterd2/xlator/options/options.go

```diff
@@ -324,10 +324,84 @@ func ValidateStr(o *Option, val string) error {
 
 // ValidateTime validates if the option is valid time format
 func ValidateTime(o *Option, val string) error {
-	if validate.IsTime(val, "hh:mm:ss") != true {
-		return ErrInvalidArg
+	var time int
+	var t string
+	var err error
+	if strings.HasSuffix(val, "sec") {
+		value := strings.TrimSuffix(val, "sec")
+		err = ValidateRange(o, value)
+		if err != nil {
+			return err
+		}
+	} else if strings.HasSuffix(val, "s") {
+		value := strings.TrimSuffix(val, "s")
+		err = ValidateRange(o, value)
```

Comment:
Is `strconv.Atoi` not needed here ?

Also, `ValidateRange` also has a call `strconv.ParseFloat`

Reply:
The valid is designed to take input argument as a string. If changing it to int is fine and wont break compatibility for other commands, then the validateRange's arguments have to be changed.

Reply:
I'm sorry, I should've been clearer. I was hinting at taking common code out of the case blocks wherever possible. Example:

```go
var tStr string
multiplier := 1

if strings.HasSuffix(val, "sec") {
    tStr = strings.TrimSuffix(val, "sec")
}
...
else if strings.HasSuffix(val, "min") {
    tStr = strings.TrimSuffix(val, "min")
    multiplier = 60
}
...
else {
    tStr = value
}

time, err := strconv.Atoi(tStr)
if err != nil {
    return err
}
time = time * multiplier
return ValidateRange(o, strconv.Itoa(time))
```

Reply:
done